server/filestore.go

```diff
@@ -229,7 +229,7 @@ func newFileStoreWithCreated(fcfg FileStoreConfig, cfg StreamConfig, created tim
 	fs := &fileStore{
 		fcfg: fcfg,
 		cfg:  FileStreamInfo{Created: created, StreamConfig: cfg},
-		fch:  make(chan struct{}),
+		fch:  make(chan struct{}, 1), // Make buffered on purpose.
```

Comment:
Could you also please check: ConsumerStore() where we create an fch but that one seem to not even be used?

You may have the same issue with consumer.go, AddConsumer() where we create fch the same way (non buffered) but use select{} to signal, which can lead to loss signal.